src/test/java/MinesweeperControllerTest.java

```diff
@@ -0,0 +1,65 @@
+import javafx.scene.layout.GridPane;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.testfx.framework.junit5.ApplicationExtension;
+import java.util.ArrayList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+
+@ExtendWith(ApplicationExtension.class)
+public class MinesweeperControllerTest {
+    private MinesweeperController controller;
+    private final int boardSize = 20;
+    private MinesweeperCell[][] board = new MinesweeperCell[boardSize][boardSize];
+    private GridPane gridPane;
+    // how to test constructor? especially since constructor calls placeMines() method
+    @Test
+    public void initialize(){
+        //given
+        givenMinesweeperController();
+
+        //when
+        controller.initialize();
+
+        //then
+        //verify cells placed on gridPane
+        //verify addOnClick method called, and works
+
+    }
+
+    @Test
+    public void getAdjacentCells(){ //passes sometimes?
+        //given
+        givenMinesweeperController();
+        ArrayList<MinesweeperCell> expectedAdjacentCells = new ArrayList<>();
+        MinesweeperCell sampleCell = new MinesweeperCell(1,0);
+        //top left off grid
+        expectedAdjacentCells.add(new MinesweeperCell(0,0)); //left
+        expectedAdjacentCells.add(new MinesweeperCell(0,1)); //bottom left
+        //top right off grid
+        expectedAdjacentCells.add(new MinesweeperCell(1,1)); //bottom middle;
+        expectedAdjacentCells.add(new MinesweeperCell(2,0)); //right
+        expectedAdjacentCells.add(new MinesweeperCell(2,1)); //bottom right
+
+
+        //when
+        ArrayList<MinesweeperCell> adjacentCells = controller.getAdjacentCells(sampleCell);
+
+        //then
+        //is this using the MinesweeperCell equals method?
```

Comment:
This is using the equals() method however your equals method checks for more than just the x,y coordinates.